### PR TITLE
docs: Improve examples/cdk/README.md

### PR DESCRIPTION
**Pull request by alex-m-aws**
## Description of your changes

Write a dedicated readme that explains how to deploy and interact with the CDK examples. Also update stack identifiers, e.g. CdkAppStack, to more specific ones, e.g. 'LambdaPowertoolsTypeScript-ExamplesCdkStack', to consider users that are deploying this stack amongst others into their accounts.

The current README is just the default CDK TS readme and is therefore not helpful.

No dependencies for this change.

### Related issues, RFCs

[#XXXXX](https://github.com/awslabs/aws-lambda-powertools-typescript/issues/XXXXX)  
[#ZZZZZ](https://github.com/awslabs/aws-lambda-powertools-typescript/issues/ZZZZZ)

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [x] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [x] Any *dependent changes have been merged and published* in downstream module
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

N/A

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/cdk/README.md [hunk 1]
```diff
@@ -1,14 +1,21 @@
-# Welcome to your CDK TypeScript project!
+# AWS Lambda Powertools (TypeScript) examples in CDK
 
-This is a blank project for TypeScript development with CDK.
+This is a deployable CDK app that deploys AWS Lambda functions as part of a CloudFormation stack. These Lambda functions use the utilities made available as part of AWS Lambda Powertools (TypeScript) to demonstrate their usage.
 
-The `cdk.json` file tells the CDK Toolkit how to execute your app.
+The example functions, located in the `lib` folder, are triggered automatically when deployed using the CDK construct defined in `lib/example-function.ts`.
```
Comment (dreamorosi) [inline examples/cdk/README.md:5]: Minor suggestions:
- replace triggered with "invoked"
- specify that they're invoked twice

The reason why I think the second point is relevant is that it'll allow customers to see different logs/metrics/annotations on segments based on when the execution had a cold start or not.

File: examples/cdk/README.md [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
- * `npm run watch`   watch for changes and compile
- * `npm run test`    perform the jest unit tests
- * `cdk deploy`      deploy this stack to your default AWS account/region
- * `cdk diff`        compare deployed stack with current state
- * `cdk synth`       emits the synthesized CloudFormation template
+ * Navigate to this location of the repo in your terminal (`examples/cdk`)
+ * `npm install`
+ * `npx cdk deploy --all --profile <YOUR_AWS_PROFILE>`
+
+Note: Prior to deploying you may need to run `npx cdk bootstrap aws://<YOU_AWS_ACCOUNT_ID>/<AWS_REGION> --profile <YOUR_AWS_PROFILE>` if you have not already bootstrapped your account for CDK.
```
Comment (dreamorosi) [inline examples/cdk/README.md:13]: Here I'd also specify that we are using CDK v2 and as such the user needs to have `cdk` installed on their machine.

File: examples/cdk/README.md [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
+
```
Comment (dreamorosi) [inline examples/cdk/README.md:14]: Here I'd also add a notice similar to the one in [`CONTRIBUTE.md`](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/CONTRIBUTING.md?plain=1#L73) about needing a valid AWS Account but most importantly about the fact that deploying this might incur in costs.

File: examples/cdk/README.md [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
+## Viewing Utility Outputs
+
+All utility outputs can be viewed from the CloudWatch console.
```
Comment (dreamorosi) [inline examples/cdk/README.md:17]:
```suggestion
All utility outputs can be viewed in the Amazon CloudWatch console.
```

File: examples/cdk/README.md [hunk 5]
```diff
@@ -1,14 +1,21 @@
...
+
+ * `Logger` outputs to Logs > Log groups
+ * `Metrics` outputs to Metrics > All metrics > LambdaPowertoolsTypeScript-CDKExample
+ * `Tracer` outputs to  X-Ray traces > Traces
```
Comment (dreamorosi) [inline examples/cdk/README.md:21]: The term "outputs" is technically incorrect as none of the utilities interacts directly with either CloudWatch or X-Ray. I'd rephrase this section to convey that these sections you mention are where one can find these outputs instead (without specifying _how_ they get there).

File: packages/metrics/tests/e2e/decorator.test.MyFunction.ts [hunk 6]
```diff
@@ -2,7 +2,7 @@ import { Metrics, MetricUnits } from '../../src';
 import { Context } from 'aws-lambda';
 import { LambdaInterface } from '../../examples/utils/lambda/LambdaInterface';
 
-const namespace = process.env.EXPECTED_NAMESPACE ?? 'CDKExample';
+const namespace = process.env.EXPECTED_NAMESPACE ?? 'LambdaPowertoolsTypeScript-CDKExample';
```
Comment (dreamorosi) [inline packages/metrics/tests/e2e/decorator.test.MyFunction.ts:5]: Any particular reason why are we applying the same change to e2e tests? I'm open to discuss the change but in a separate PR, so to keep this one focused on examples.

File: packages/metrics/tests/e2e/standardFunctions.test.MyFunction.ts [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import { Metrics, MetricUnits } from '../../src';
 import { Context } from 'aws-lambda';
 
-const namespace = process.env.EXPECTED_NAMESPACE ?? 'CDKExample';
+const namespace = process.env.EXPECTED_NAMESPACE ?? 'LambdaPowertoolsTypeScript-CDKExample';
```
Comment (dreamorosi) [inline packages/metrics/tests/e2e/standardFunctions.test.MyFunction.ts:4]: Same as other file, I'm open to discuss the change but in a separate PR, so to keep this one focused on examples.

**Review by dreamorosi — CHANGES_REQUESTED** [review]

File: examples/cdk/README.md (hunk 2 above)
Comment (alex-m-aws) [reply examples/cdk/README.md:13]: okay so ditch npx and assert an installed CDK v2

File: packages/metrics/tests/e2e/decorator.test.MyFunction.ts (hunk 6 above)
Comment (alex-m-aws) [reply packages/metrics/tests/e2e/decorator.test.MyFunction.ts:5]: Because the Metric examples are found in Metrics > All metrics > CdkExample.
Better to have a more specific name for users that may have other CDK stacks in their account, though maybe unnecesary to change in tests... couldn't find where else CdkExample was coming from.

**Comment by alex-m-aws** [comment]
@dreamorosi comments addressed in de85af38fedaeb5b987862cf490ea7e8dfbd87ec

File: packages/metrics/tests/e2e/decorator.test.MyFunction.ts (hunk 6 above)
Comment (dreamorosi) [reply packages/metrics/tests/e2e/decorator.test.MyFunction.ts]: Yea agree that we should find a more specific name for these metrics' namespace but this stack in the tests is completely unrelated from the one in examples so I'd be more keen to make this change in another PR. In any case happy to hear also the opinion of a second reviewer, let's leave the conversation open.

File: examples/cdk/README.md [hunk 8]
```diff
@@ -1,14 +1,24 @@
-# Welcome to your CDK TypeScript project!
+# AWS Lambda Powertools (TypeScript) examples in CDK
 
-This is a blank project for TypeScript development with CDK.
+This is a deployable CDK app that deploys AWS Lambda functions as part of a CloudFormation stack. These Lambda functions use the utilities made available as part of AWS Lambda Powertools (TypeScript) to demonstrate their usage.
 
-The `cdk.json` file tells the CDK Toolkit how to execute your app.
+You will need to have a valid AWS Account in order to deploy these resources. These resources may incur costs to your AWS Account.
```
Comment (dreamorosi) [inline examples/cdk/README.md:5]:
```suggestion
You will need to have a valid AWS Account in order to deploy these resources. These resources may incur costs to your AWS Account. The cost from **some services** are covered by the [AWS Free Tier](https://aws.amazon.com/free/?all-free-tier.sort-by=item.additionalFields.SortRank&all-free-tier.sort-order=asc&awsf.Free%20Tier%20Types=*all&awsf.Free%20Tier%20Categories=*all) but not all of them. If you don't have an AWS Account follow [these instructions to create one](https://aws.amazon.com/premiumsupport/knowledge-center/create-and-activate-aws-account/).
```

**Review by dreamorosi — APPROVED** [review]

**Review by flochaz — APPROVED** [review]
